app/src/main/cpp/skyline/services/socket/bsd/IClient.cpp

```diff
@@ -1,84 +1,253 @@
+        i32 result{getsockopt(fd, level, GetOption(optionName), request.outputBuf.at(0).data(), &addrLen)};
+        return PushBsdResult(response, 0, errno);
     }
 
     Result IClient::Listen(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        return {};
+        i32 fd{request.Pop<i32>()};
+        i32 backlog{request.Pop<i32>()};
+        i32 result{listen(fd, backlog)};
+        return PushBsdResult(response, 0, errno);
+    }
+
+    Result IClient::Fcntl(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        i32 fd{request.Pop<i32>()};
+        i32 cmd{request.Pop<i32>()};
+        i32 arg{request.Pop<i32>()};
+        i32 result{fcntl(fd, cmd, arg)};
+        return PushBsdResult(response, 0, errno);
     }
 
     Result IClient::SetSockOpt(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        return {};
+        i32 fd{request.Pop<i32>()};
+        i32 level{request.Pop<i32>()};
+        OptionName optionName{request.Pop<OptionName>()};
+        if (level == 0xFFFF)
+            level = SOL_SOCKET;
+        i32 result{setsockopt(fd, level, GetOption(optionName), request.inputBuf.at(0).data(), static_cast<socklen_t>(request.inputBuf.at(0).size()))};
+        return PushBsdResult(response, 0, errno);
     }
 
     Result IClient::Shutdown(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        return {};
+        i32 fd{request.Pop<i32>()};
+        i32 how{request.Pop<i32>()};
+        i32 result{shutdown(fd, how)};
+        return PushBsdResult(response, 0, errno);
     }
 
     Result IClient::ShutdownAllSockets(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
         return {};
     }
 
     Result IClient::Write(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        return {};
+        i32 fd{request.Pop<i32>()};
+        i32 flags{request.Pop<i32>()};
+        ssize_t result{send(fd, request.inputBuf.at(0).data(), request.inputBuf.at(0).size(), flags)};
+        return PushBsdResultErrno(response, result);
     }
 
     Result IClient::Read(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        response.Push<u32>(0);
-        response.Push<u32>(0);
-        return {};
+        i32 fd{request.Pop<i32>()};
+        ssize_t result{recv(fd, request.outputBuf.at(0).data(), request.outputBuf.at(0).size(), 0)};
+        return PushBsdResultErrno(response, result);
     }
 
     Result IClient::Close(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        i32 fd{request.Pop<i32>()};
+        i32 result{close(fd)};
+        return PushBsdResult(response, 0, errno);
+    }
+
+    Result IClient::EventFd(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        return PushBsdResult(response, 1, 0);
```

Comment:
this needs to be implemented or print error at min, same for any other func thats stubbed but needs to actually do smth